src/dc_receive_imf.rs

```diff
@@ -1893,28 +1893,17 @@ unsafe fn dc_is_reply_to_known_message(
 }
 
 unsafe fn is_known_rfc724_mid_in_list(context: &Context, mid_list: *const clist) -> bool {
-    if !mid_list.is_null() {
-        let mut cur: *mut clistiter;
-        cur = (*mid_list).first;
-        while !cur.is_null() {
-            if 0 != is_known_rfc724_mid(
-                context,
-                (if !cur.is_null() {
-                    (*cur).data
-                } else {
-                    ptr::null_mut()
-                }) as *const libc::c_char,
-            ) {
-                return true;
-            }
-            cur = if !cur.is_null() {
-                (*cur).next
-            } else {
-                ptr::null_mut()
-            }
+    if mid_list.is_null() {
+        return false;
+    }
+
+    for data in &*mid_list {
+        if 0 != is_known_rfc724_mid(context, data.cast()) {
+            return true;
```

Comment:
Maybe use [any](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any)?

Reply:
@link2xt In next PR.